### ISSUE TITLE
Expected valid bigint: 0 < bigint < curve.n

### ISSUE THREAD

**Issue by smakosh** [apibara/starknet-react#510]
After upgrading to `3.0.1-beta.0` we started getting the following error: "Expected valid bigint: 0 < bigint < curve.n"

Here's the code responsible for that error:
```ts
const onSubmit = async ({ amount }: DepositFormData) => {
    try {
      if (!account || !assetContract) return
      setDisabled(true)
      
      const preAllowance = await assetContract.allowance(
        account.address,
        CONTROLLER
      )
      const txs = []

      if (
        uint256.uint256ToBN(
          preAllowance || {
            low: 0,
            high: 0,
          }
        ) < BigInt(parseUnits(amount.toString(), DECIMALS))
      ) {
        txs.push({
          contractAddress: asset,
          entrypoint: 'approve',
          calldata: [
            CONTROLLER,
            uint256.bnToUint256(parseUnits(amount.toString(), DECIMALS)).low,
            uint256.bnToUint256(parseUnits(amount.toString(), DECIMALS)).high,
          ],
        })
      }

      txs.push({
        contractAddress: CONTROLLER,
        entrypoint: 'deposit_collateral',
        calldata: [
          asset,
          uint256.bnToUint256(parseUnits(amount.toString(), DECIMALS)).low,
          uint256.bnToUint256(parseUnits(amount.toString(), DECIMALS)).high,
        ],
      })

      const depositTx = await account.execute(txs)

      setTxHash(depositTx.transaction_hash)
      setTxStatus({
        success: 'Collateral deposited',
        loading: 'Transaction pending',
        error: 'Failed to deposit collateral',
      })
    } catch (e: unknown) {
      toast.error((e as { toString: () => string }).toString())
    } finally {
      setDisabled(false)
    }
  }
```

**Comment by fracek** [comment]
Can you provide the following information:

 * a stack trace
 * which wallet you are using

Also try to `console.log` all bigint values and see if anything looks suspcious.

**Comment by smakosh** [comment]
> Can you provide the following information:
> 
> * a stack trace
> * which wallet you are using
> 
> Also try to `console.log` all bigint values and see if anything looks suspcious.

I'm using Argent X.

![image](https://github.com/user-attachments/assets/4d9c2987-6f64-4f1b-bf63-f527756d56a9)

**Comment by jancris100** [comment]
Can you assign this to me?

**Comment by smakosh** [comment]
for more details @fracek 

Imports

```ts
import {
  useAccount,
  useConnect,
  useContract,
  useProvider,
  useTransactionReceipt,
} from '@starknet-react/core'
import { uint256 } from 'starknet'
import { useStarknetkitConnectModal } from 'starknetkit'
```

```ts
const { connect, connectors } = useConnect()
  const { starknetkitConnectModal } = useStarknetkitConnectModal({
    connectors: connectors as any,
  })
  
const connectWallet = async () => {
    const { connector } = await starknetkitConnectModal()
    if (connector) {
      connect({ connector })
    }
  }
```
```tsx
'use client'

import { sepolia } from '@starknet-react/chains'
import { publicProvider, StarknetConfig, starkscan } from '@starknet-react/core'
import { constants } from 'starknet'
import {
ArgentMobileConnector,
isInArgentMobileAppBrowser,
} from 'starknetkit/argentMobile'
import { InjectedConnector } from 'starknetkit/injected'
import { WebWalletConnector } from 'starknetkit/webwallet'

const chains = [sepolia]

export const availableConnectors = isInArgentMobileAppBrowser()
? [
    ArgentMobileConnector.init({
      options: {
        url: typeof window !== 'undefined' ? window.location.href : '',
        dappName: 'Company name',
        chainId: constants.NetworkName.SN_SEPOLIA,
      },
    }),
  ]
: [
    new InjectedConnector({ options: { id: 'argentX' } }),
    new InjectedConnector({ options: { id: 'braavos' } }),
    ArgentMobileConnector.init({
      options: {
        url: typeof window !== 'undefined' ? window.location.href : '',
        dappName: 'Company name',
        chainId: constants.NetworkName.SN_SEPOLIA,
      },
    }),
    new WebWalletConnector({ url: process.env.NEXT_PUBLIC_WEB_WALLET_URL }),
  ]

export function StarknetProvider({ children }: { children: React.ReactNode }) {
return (
  <StarknetConfig
    chains={chains}
    provider={publicProvider()}
    connectors={availableConnectors}
    explorer={starkscan}
    autoConnect
  >
    {children}
  </StarknetConfig>
)
}
```

**Comment by ScottyDavies** [comment]
Hello @smakosh can i please be assigned this task?

**Comment by smakosh** [comment]
> Hello @smakosh can i please be assigned this task?

I'm not the owner/maintainer of the repo, cannot assign issues here

**Comment by RedBeardEth** [comment]
![image](https://github.com/user-attachments/assets/ac5cecf5-5abf-4ebb-83d7-c621afc1ef89)

@fracek This is the trace stack - and is only occurring when attempting to use account.execute()
![image](https://github.com/user-attachments/assets/464bf24d-e928-414f-9a3d-5c6d4243c112)
Not sure if related - but the account pk is 0

**Comment by fracek** [comment]
Looks like the `Signer` object Which delegates to `@scure/starknet` (which in turn uses `noble-curves`) checks that the [private key is in range](https://github.com/paulmillr/noble-curves/blob/main/src/abstract/weierstrass.ts#L293-L313) (defined as `[1, N)`) and the key being 0 doesn't pass this check.

Can you check if the [following demo](https://www.starknet-react.com/demo/send-transaction) works for you using ArgentX?

**Comment by smakosh** [comment]
> following demo

Unable to connect the wallet, account address loads forever
<img width="815" alt="Screenshot 2024-10-07 at 9 05 38 AM" src="https://github.com/user-attachments/assets/fcc708f9-0345-4ee4-a3e0-c9a08e463c2c">

**Comment by fracek** [comment]
Does it print any log? Can you download this repo and run the demo locally and try to see what's happening?

**Comment by smakosh** [comment]
> Does it print any log? Can you download this repo and run the demo locally and try to see what's happening?

Ran it locally, here's what I get on the log

```
Uncaught TypeError: Cannot read properties of undefined (reading 'id')
```

**Comment by smakosh** [comment]
Also this demo doesn't use the account.execute() method

**Comment by fracek** [comment]
I noticed you're using starknetkit. Are you using the latest release (or prerelease) that is compatible with starknet react v3? V3 switches to the new wallet API to execute transactions.

**Comment by RedBeardEth** [comment]
This is using   "starknetkit": "^2.3.2" (latest)

**Comment by fracek** [comment]
That's strange because it's the same version we use in the demo.

Last thing that could differ is the browser/OS you use and the argent version. Also did you test it with Braavos? Does it work with it?

**Comment by kseikyo** [comment]
### TLDR: Downgrade to `"starknet-react: 3.0.0"` and `"starknetkit": "^2.3.0"` for now.


Had the same error as op happen once after upgrading
```diff
- "starknet-react: 3.0.0",
+ "starknet-react: 3.0.1",
- "starknetkit": "^2.3.0",
+ "starknetkit": "^2.3.2",
```

Using starknet-react hooks to get the account and provider to use in the following code results in the error below:
```ts
const contract = new Contract(ABI, address, provider).typedv2(ABI);
console.log(contract); // logs correctly

const tx = contract.populateTransaction.fn(params);

return account.execute(tx); // will throw
```

```
LibraryError: RPC: starknet_getNonce with params {
  "contract_address": "0x68f5ed041a0fad85ad85d8f7b6653abd806ebe6e80b92faf501a883589347b7",
  "block_id": "pending"
}
 
        20: Contract not found: undefined
    at RpcChannel2.errorHandler (index.mjs:4134:13)
    at RpcChannel2.fetchEndpoint (index.mjs:4151:12)
    at async Account.execute (index.mjs:6464:45)
```

**Comment by smakosh** [comment]
I resolved my issue by moving away from `account.execute()` to `useSendTransaction`

Example:

```ts
const { sendAsync: doSomething } = useSendTransaction({
  calls: undefined,
})

const onSubmit = async () => {
  try {
    const { transaction_hash } = await doSomething([
      {
        contractAddress,
        entrypoint: 'some_function_name',
        calldata: [
          somedata,
          otherdata
        ],
      },
      {
        contractAddress,
        entrypoint: 'some_other_function_name',
        calldata: [
         somedata,
          otherdata
        ],
      },
    ])
    
    // Do something with `transaction_hash`
  } catch (e) {} 
}
```

**Comment by fracek** [comment]
Does `account` come from `useAccount`? Because that object is from starknet.js so it may be a bug in that library.

**Comment by smakosh** [comment]
> Does `account` come from `useAccount`? Because that object is from starknet.js so it may be a bug in that library.

`useAccount` is imported from starknet-react

**Comment by fracek** [comment]
Yes, but the `account` [object returned](https://github.com/apibara/starknet-react/blob/main/packages/core/src/connectors/injected.ts#L112-L123) is `WalletAccount` from starknet.js.

**Comment by smakosh** [comment]
Closing this for now, see https://github.com/apibara/starknet-react/issues/510#issuecomment-2407675745 if you want to solve this
